qiskit_nature/utils/__init__.py

```diff
@@ -0,0 +1,32 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Utilities (:mod:`qiskit_nature.utils`)
+==============================================
+
+.. currentmodule:: qiskit_nature.utils
+
+
+.. autosummary::
+   :toctree:
+"""
+
+from qiskit_nature.utils.linalg import (
```

Comment:
linalg also has this as a public function `fermionic_gaussian_decomposition_jw` is there a reason we do not have it here too?

Maybe this is too premature but assuming this utils grows and we have different modules in here with different types of functions would it be better to have a Linear Algebra section and list the functions as a table, like we have for the classes elsewhere, where it links over to the detail. As it stands the functions come out together on the one page so it will get busy very quickly

Reply:
I don't want to expose `fermionic_gaussian_decomposition_jw` because its output is somewhat opaque. It's better thought of as an internal helper function.

It makes sense to have `linalg` be a separate module and documented as such. But I would advocate for it being at the top level, i.e. `qiskit_nature.linalg` because this mirrors what Numpy and Scipy have, so it would be more easily discovered by users. What do you think? If you disagree then I can just it a new module under `utils`, but in my opinion this nested structure is unnecessarily convoluted.

Reply:
I think things are ok in terms of structure. I.e utils sub-package and linalg.py module. I can imagine utils gathering different useful modules. I am not sure we have enough linalg that we want to expose things as a sub-package. Its not that we do this like numpy/scipy where its core function where they have a bunch and it makes sense; for us its more supporting logic right that a normal user of our API doing Nature will probably never go to, though they can, I think its mainly for developers of Nature code right.

I was more looking at the html output and the way how we normally do this. Here for instance is [Terra utils sub-package documentation](https://qiskit.org/documentation/apidoc/utils.html). As you can see the functions are in a table (as we normally do on these pages with classes). And they are separated into separate tables as a logical partitioning rather than throwing them all into one big one. That's all I meant, where the page says these are utilities or whatever and then there is a section header saying linear algebra with the two functions listed under it so they end up formatted as a table that has the summary of function there and link to the full doc (just like the output you see elsewhere where we list things)

Reply:
Done.